src/ring_theory/non_zero_divisors.lean

```diff
@@ -39,42 +39,65 @@ begin
     rw [mul_assoc, hx] },
 end
 
-lemma eq_zero_of_ne_zero_of_mul_right_eq_zero {x y : A} (hnx : x ≠ 0) (hxy : y * x = 0) : y = 0 :=
+lemma eq_zero_of_ne_zero_of_mul_right_eq_zero [no_zero_divisors M]
+  {x y : M} (hnx : x ≠ 0) (hxy : y * x = 0) : y = 0 :=
 or.resolve_right (eq_zero_or_eq_zero_of_mul_eq_zero hxy) hnx
 
-lemma eq_zero_of_ne_zero_of_mul_left_eq_zero {x y : A} (hnx : x ≠ 0) (hxy : x * y = 0) : y = 0 :=
+lemma eq_zero_of_ne_zero_of_mul_left_eq_zero [no_zero_divisors M]
+  {x y : M} (hnx : x ≠ 0) (hxy : x * y = 0) : y = 0 :=
 or.resolve_left (eq_zero_or_eq_zero_of_mul_eq_zero hxy) hnx
 
-lemma mem_non_zero_divisors_iff_ne_zero {x : A} : x ∈ non_zero_divisors A ↔ x ≠ 0 :=
+lemma mem_non_zero_divisors_iff_ne_zero [no_zero_divisors M] [nontrivial M] {x : M} :
+  x ∈ non_zero_divisors M ↔ x ≠ 0 :=
 ⟨λ hm hz, zero_ne_one (hm 1 $ by rw [hz, one_mul]).symm,
-λ hnx z, eq_zero_of_ne_zero_of_mul_right_eq_zero hnx⟩
+ λ hnx z, eq_zero_of_ne_zero_of_mul_right_eq_zero hnx⟩
 
-lemma map_ne_zero_of_mem_non_zero_divisors [nontrivial R] {B : Type*} [ring B] {g : R →+* B}
-  (hg : function.injective g) {x : R} (h : x ∈ non_zero_divisors R) : g x ≠ 0 :=
+lemma map_ne_zero_of_mem_non_zero_divisors [nontrivial M]
```

Comment:
I think for consistency it probably makes sense to call this
```suggestion
lemma monoid_with_zero_hom.map_ne_zero_of_mem_non_zero_divisors [nontrivial M]
```
and make the `g` argument explicit to enable dot notation.